src/Analyzers/Core/Analyzers/IDEDiagnosticIdToOptionMappingHelper.cs

```diff
@@ -20,48 +20,27 @@ namespace Microsoft.CodeAnalysis.Diagnostics
     internal static class IDEDiagnosticIdToOptionMappingHelper
     {
         private static readonly ConcurrentDictionary<string, ImmutableHashSet<IOption2>> s_diagnosticIdToOptionMap = new();
-        private static readonly ConcurrentDictionary<string, ConcurrentDictionary<string, ImmutableHashSet<IOption2>>> s_diagnosticIdToLanguageSpecificOptionsMap = new();
```

Comment:
This will not work - we need a different per-language map as there are distinct options used by certain C# and VB analyzers that share the same diagnostic ID and base type. In fact, my first implementation did not have this second dictionary, but that led to only C# or VB option being stored for the diagnostic ID, but we need to store both for some cases.

Reply:
Yes, that's what I observed. I think an alternative is to add `LanguageName` property to `ILanguageSpecificOption`, especially because I see

https://github.com/dotnet/roslyn/blob/2c86eb91e511971fcf721f53a6efa4fdac4a57b6/src/EditorFeatures/Core/EditorConfigSettings/Updater/SettingsUpdateHelper.cs#L90

which looks incorrect, and the correct implementation can make use of the suggested new property.

@mavasani Let me know what do you think of this.